test/parallel/test-fs-whatwg-url.js

```diff
@@ -72,3 +72,10 @@ if (common.isWindows) {
     message: `File URL host must be "localhost" or empty on ${os.platform()}`
   }));
 }
+
+// path ending with forward slash are not permitted
+fs.readFile(new URL('file:///c:/tmp/test/'), common.expectsError({
```

Comment:
Can you add tests for other APIs with the validation changed as well? Also it would be more robust if we test paths like `file:///tmp/test/` on non-windows systems because `file:///c:/tmp/test/` is obviously invalid there for a different reason(drives).

Reply:
You can test different type of URLs on different systems by reusing the tested URL object like this:

```js
const urlWithSlash = common.isWindows ?
  new URL('file:///c:/tmp/test/') : new URL('file:///tmp/test/');

fs.readFile(urlWithSlash, common.expectsError(...));
```